### PR TITLE
refactor: update `blas/ext/base/dnanasum` to follow current project conventions

### LINKED ISSUE #1476: [RFC]: refactor `blas/ext/base/dnanasum` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/dnanasum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/dnanasum) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by raunak-dev-edu**
Resolves #1476 .

## Description

> What is the purpose of this pull request?

This pull request:

-   refactors `blas/ext/base/dnanasum` to follow current project conventions.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1476
-   progress #1152

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dnanasum/benchmark/benchmark.js [hunk 1]
```diff
@@ -39,18 +40,15 @@ var dnanasum = require( './../lib/dnanasum.js' );
 * @returns {Function} benchmark function
 */
 function createBenchmark( len ) {
-	var x;
-	var i;
+	var x = filledarrayBy( len, 'float64', clbk );
+	return benchmark;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnanasum/benchmark/benchmark.js:44]: why are you returning benchmark here?

File: lib/node_modules/@stdlib/blas/ext/base/dnanasum/benchmark/benchmark.native.js [hunk 2]
```diff
@@ -48,19 +49,15 @@ var opts = {
 * @returns {Function} benchmark function
 */
 function createBenchmark( len ) {
-	var x;
-	var i;
+	var x = filledarrayBy( len, 'float64', clbk );
+	return benchmark;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnanasum/benchmark/benchmark.native.js:53]: same comment

File: lib/node_modules/@stdlib/blas/ext/base/dnanasum/manifest.json [hunk 3]
```diff
@@ -35,8 +35,12 @@
 			],
 			"libpath": [],
 			"dependencies": [
+				"@stdlib/napi/export",
+                "@stdlib/napi/argv",
+                "@stdlib/napi/argv-int64",
+                "@stdlib/napi/argv-strided-float64array",
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/dnanasum/manifest.json]: incorrect indentation

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Left a few comments, and questions. This PR needs work.

File: lib/node_modules/@stdlib/blas/ext/base/dnanasum/benchmark/benchmark.js (hunk 1 above)
Comment (raunak-dev-edu) [reply lib/node_modules/@stdlib/blas/ext/base/dnanasum/benchmark/benchmark.js:44]: It seems to confuse me. Is it not returning benchmark here. To update this package, I have to use a utility function for generating random numbers and remove the loop. I've done that, and the benchmark is returning as it was before. I only made changes to the for loop part, which was  
```
x = new Float64Array( len );
for ( i = 0; i < x.length; i++ ) {
	if ( randu() < 0.2 ) {
		x[ i ] = NaN;
	} else {
		x[ i ] = ( randu()*20.0 ) - 10.0;
}
```

File: lib/node_modules/@stdlib/blas/ext/base/dnanasum/manifest.json (hunk 3 above)
Comment (raunak-dev-edu) [reply lib/node_modules/@stdlib/blas/ext/base/dnanasum/manifest.json]: I have update the manifest with correct indentation

**Review by Planeshifter — APPROVED** [review]
Did some minor clean-up. Should be ready to merge now. Thanks @raunak-dev-edu!

**Pull request merged**